subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultCacheLockingManager.java

```diff
@@ -39,9 +47,22 @@ public DefaultCacheLockingManager(CacheRepository cacheRepository, ArtifactCache
                 .withCrossVersionCache(CacheBuilder.LockTarget.CacheDirectory)
                 .withDisplayName("artifact cache")
                 .withLockOptions(mode(FileLockManager.LockMode.None)) // Don't need to lock anything until we use the caches
+                .withCleanup(createCleanupAction(cacheMetaData))
```

Comment:
I presume that the exclusive locking done here will prevent the cleanup from occurring whenever another build is using the dependency cache in any way. However, I'm not familiar enough with the locking to be sure of this. You'll want to check!

Reply:
Yes, the cleanup is run with exclusive locking.